test/virtual/include/openenclave/edger8r/common.h

```diff
@@ -66,35 +66,46 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
     return result;
 }
 
-#define OE_ADD_SIZE(total, size)                                   \
-    do                                                             \
-    {                                                              \
-        if (sizeof(total) > sizeof(size_t) && total > OE_SIZE_MAX) \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (sizeof(size) > sizeof(size_t) && size > OE_SIZE_MAX)   \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (oe_add_size((size_t*)&total, (size_t)size) != OE_OK)   \
-        {                                                          \
-            _result = OE_INTEGER_OVERFLOW;                         \
-            goto done;                                             \
-        }                                                          \
+#define OE_MUL_SIZE(a, b, c)                         \
+    do                                               \
+    {                                                \
+        if ((a) && (b) && ((a) > OE_SIZE_MAX / (b))) \
```

Comment:
Do we need to use __a , __b local variables?
Also do we need to perform the check for b > OE_SIZE_MAX/a ?

Reply:
Used local for `a` and `b`.
There is no need for checking b > OE_SIZE_MAX/a, which is equivalent to the a > OE_SIZE_MAX/b.